src/test/scala/chiselTests/naming/IdentifierProposerSpec.scala

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests
+package naming
+
+import chisel3._
+import chisel3.naming.IdentifierProposer
+
+case class Blah(i: Int)
+class Foo(val i: Int)
+object Foo {
+  implicit val fooProposer = new IdentifierProposer[Foo] {
+    def propose(foo: Foo): String = s"Foo(${foo.i})"
+  }
+}
+class Baz(val i: Int)
+
+class IdentifierProposerSpec extends ChiselFunSpec with Utils {
+  it("(0): getProposal works on a variety of object types") {
+    assert(IdentifierProposer.getProposal("Hi") == "Hi")
+    assert(IdentifierProposer.getProposal(1) == "1")
+    assert(IdentifierProposer.getProposal(Blah(1)) == "Blah_1")
+    assert(IdentifierProposer.getProposal(new Bundle {}) == "AnonymousBundle")
+    assert(IdentifierProposer.getProposal(new Foo(1)) == "Foo_1")
```

Comment:
shouldn't this be `Foo(1)` given the above proposer?

Reply:
It gets filtered to be a clean name without "(", see `def filterProposal` on `object IdentiferProposer`